### ISSUE TITLE
scene.camera crashes

### ISSUE THREAD

**Issue by sariug** [mikedh/trimesh#594]
I tried the code around 150 different files. just 1 of them crushes and I do not know why. 

The error message is:

> Traceback (most recent call last):
  File "/snap/pycharm-community/163/helpers/pydev/_pydevd_bundle/pydevd_exec.py", line 3, in Exec
    exec exp in global_vars, local_vars
  File "<input>", line 1, in <module>
  File "/home/ugurcan/Desktop/GitBridge/cad-feature-extraction/venv/local/lib/python2.7/site-packages/trimesh/scene/scene.py", line 542, in camera
    return self.set_camera()
  File "/home/ugurcan/Desktop/GitBridge/cad-feature-extraction/venv/local/lib/python2.7/site-packages/trimesh/scene/scene.py", line 458, in set_camera
    self.bounds_corners,
  File "/home/ugurcan/Desktop/GitBridge/cad-feature-extraction/venv/local/lib/python2.7/site-packages/trimesh/caching.py", line 101, in get_cached
    value = function(*args, **kwargs)
  File "/home/ugurcan/Desktop/GitBridge/cad-feature-extraction/venv/local/lib/python2.7/site-packages/trimesh/scene/scene.py", line 254, in bounds_corners
    for k, v in self.geometry.items()}
  File "/home/ugurcan/Desktop/GitBridge/cad-feature-extraction/venv/local/lib/python2.7/site-packages/trimesh/scene/scene.py", line 254, in <dictcomp>
    for k, v in self.geometry.items()}
  File "/home/ugurcan/Desktop/GitBridge/cad-feature-extraction/venv/local/lib/python2.7/site-packages/trimesh/caching.py", line 101, in get_cached
    value = function(*args, **kwargs)
  File "/home/ugurcan/Desktop/GitBridge/cad-feature-extraction/venv/local/lib/python2.7/site-packages/trimesh/base.py", line 520, in bounds
    mesh_bounds = np.array([in_mesh.min(axis=0),
  File "/home/ugurcan/Desktop/GitBridge/cad-feature-extraction/venv/local/lib/python2.7/site-packages/numpy/core/_methods.py", line 29, in _amin
    return umr_minimum(a, axis, None, out, keepdims)
ValueError: zero-size array to reduction operation minimum which has no identity

The code: 
`    glb_scene = trimesh.scene.Scene()`
`for loop create some mesh:`
    `       mesh = trimesh.Trimesh(vertices,
                               faces)`
    `        glb_scene.add_geometry(mesh)`
`for loop finish`

`glb_scene.camera._resolution = (640, 480)` 
Crursh !

without this if i call export, also crushes. Whatever I do with camera, it crushes. Any opinion ?

**Comment by mikedh** [comment]
Is it possible that mesh is empty? What is `mesh.vertices.shape` and `mesh.faces.shape`?

Also, why set `camera._resolution` instead of `camera.resolution`?

**Comment by mikedh** [comment]
Hey, that crash really looks like the crash you get when there are no faces:
```
In [6]: a = trimesh.Trimesh([], [])                                  

In [7]: a.bounds                                                     
---------------------------------------------------------------------
ValueError                          Traceback (most recent call last)
<ipython-input-7-fb749b14822d> in <module>
----> 1 a.bounds

/media/psf/Dropbox/robotics/trimesh/trimesh/caching.py in get_cached(*args, **kwargs)
     99             return self._cache.cache[name]
    100         # value not in cache so execute the function
--> 101         value = function(*args, **kwargs)
    102         tic[2] = time.time()
    103         # store the value

/media/psf/Dropbox/robotics/trimesh/trimesh/base.py in bounds(self)
    518         in_mesh = self.vertices[self.referenced_vertices]
    519         # get mesh bounds with min and max
--> 520         mesh_bounds = np.array([in_mesh.min(axis=0),
    521                                 in_mesh.max(axis=0)])
    522         # should not be mutable

~/conda/lib/python3.7/site-packages/numpy/core/_methods.py in _amin(a, axis, out, keepdims, initial)
     30 def _amin(a, axis=None, out=None, keepdims=False,
     31           initial=_NoValue):
---> 32     return umr_minimum(a, axis, None, out, keepdims, initial)
     33 
     34 def _sum(a, axis=None, dtype=None, out=None, keepdims=False,

ValueError: zero-size array to reduction operation minimum which has no identity
```

**Comment by mikedh** [comment]
[This PR](https://github.com/mikedh/trimesh/pull/595/files#diff-2be5e3c65ab250c84075d43926fac586R262-R280) changes `mesh.bounds` to return None for empty meshes rather than crashing.

**Comment by mikedh** [comment]
Should be fixed by #596, though feel free free to re-open if not!

**Comment by sariug** [comment]
Finally found the bug. I was adding 300 meshes. 1 mesh was actually had 0 facet. Thanks !
